src/solvers/strings/string_constraint_generator.h

```diff
@@ -74,392 +74,248 @@ class string_constraint_generatort final
-
-string_constraintst add_constraint_on_characters(
-  symbol_generatort &fresh_symbol,
-  const array_string_exprt &s,
-  const exprt &start,
-  const exprt &end,
-  const std::string &char_set,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst> add_axioms_for_constrain_characters(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool);
 
-// The following functions add axioms for the returned value
-// to be equal to the result of the function given as argument.
-// They are not accessed directly from other classes: they call
-// `add_axioms_for_function_application` which determines which of
-// these methods should be called.
-
-std::pair<exprt, string_constraintst> add_axioms_for_char_at(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst> add_axioms_for_code_point_at(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst> add_axioms_for_code_point_before(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst> add_axioms_for_contains(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst> add_axioms_for_equals(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst> add_axioms_for_equals_ignore_case(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool);
-
-std::pair<exprt, string_constraintst> add_axioms_for_is_empty(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst> add_axioms_for_is_prefix(
-  symbol_generatort &fresh_symbol,
-  const array_string_exprt &prefix,
-  const array_string_exprt &str,
-  const exprt &offset);
-std::pair<exprt, string_constraintst> add_axioms_for_is_prefix(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  bool swap_arguments,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst> add_axioms_for_is_suffix(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  bool swap_arguments,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst> add_axioms_for_length(
-  const function_application_exprt &f,
-  array_poolt &array_pool);
-std::pair<exprt, string_constraintst>
-add_axioms_for_empty_string(const function_application_exprt &f);
-
-std::pair<exprt, string_constraintst> add_axioms_for_copy(
-  symbol_generatort &fresh_symbol,
-  const function_application_exprt &f,
-  array_poolt &array_pool);
+public:
+  std::pair<exprt, string_constraintst> add_axioms_for_concat(
```

Comment:
:bulb: at some point we should remove the `add_` from these method names, since it doesn't add them but return them (doesn't need to be in this PR)